src/v/cluster/shard_table.h

```diff
@@ -23,37 +23,81 @@ namespace cluster {
 /// \brief this is populated by consensus::controller
 /// every core will have a _full_ copy of all indexes
 class shard_table final {
+    struct shard_revision {
+        ss::shard_id shard;
+        model::revision_id revision;
+    };
+
 public:
     bool contains(const raft::group_id& group) {
         return _group_idx.find(group) != _group_idx.end();
     }
     ss::shard_id shard_for(const raft::group_id& group) {
-        return _group_idx.find(group)->second;
+        return _group_idx.find(group)->second.shard;
     }
 
     /**
      * \brief Lookup the owning shard for an ntp.
      */
     std::optional<ss::shard_id> shard_for(const model::ntp& ntp) {
         if (auto it = _ntp_idx.find(ntp); it != _ntp_idx.end()) {
-            return it->second;
```

Comment:
this is a bug, in that this will return the `original` NTP shard, not the latest? 

how come this isn't 

`ntp -> small_vector[shards]` so the version is implicitly the last item in the vector?

Reply:
@senior7515 It is not a vector, it just a tuple `shard + revision` it is kept up to date, we do not need history